modules/videoio/test/test_audio.cpp

```diff
@@ -0,0 +1,274 @@
+                {
+                    f = ((double) audioFrame.at<signed short>(0,i)) / (double) 32768;
+                    audioData.push_back(f);
+                }
+            }
+            else
+            {
+                break;
+            }
+        }
+        ASSERT_FALSE(audioData.empty());
+    }
+    void checkAudio()
+    {
+        for(unsigned int i = 0; i < validAudioData.size(); i++)
+        {
+            EXPECT_LE(fabs(validAudioData[i] - audioData[i]), epsilon) << "sample index " << i;
+        }
+    }
+};
+
+const param audioParams[] =
+{
+    param("wav", 0.0001, {"CAP_MSMF", cv::CAP_MSMF}),
+    param("mp3", 0.1, {"CAP_MSMF", cv::CAP_MSMF}),
+    param("mp4", 0.15, {"CAP_MSMF", cv::CAP_MSMF})
+};
+
+class Audio : public AudioTestFixture{};
+
+TEST_P(Audio, audio)
+{
+    if (!videoio_registry::hasBackend(cv::VideoCaptureAPIs(backend.second)))
+        throw SkipTestException(backend.first + " backend was not found");
+
+    doTest();
+}
+
+INSTANTIATE_TEST_CASE_P(/**/, Audio, testing::ValuesIn(audioParams));
+
+class MediaTestFixture : public baseAudio, public testing::TestWithParam <paramCombination>
+{
+public:
+    MediaTestFixture():
+        videoType(get<2>(GetParam())),
+        height(get<3>(GetParam())),
+        width(get<4>(GetParam())),
+        numberOfFrames(get<5>(GetParam())),
+        fps(get<6>(GetParam())),
+        psnrThreshold(get<7>(GetParam()))
+        {
+            format = get<0>(GetParam());
+            epsilon = get<1>(GetParam());
+            backend = get<8>(GetParam());
+            root = "audio/";
+            fileName = "test_audio";
+            params = {  CAP_PROP_AUDIO_STREAM, 0,
+                        CAP_PROP_VIDEO_STREAM, 0,
+                        CAP_PROP_AUDIO_DATA_DEPTH, CV_16S };
+        };
+    void doTest()
+    {
+        getValidAudioData();
+        getValidVideoData();
+        getFileData();
+        comparisonAudio();
+        comparisonVideo();
+    }
+
+private:
+    void getValidAudioData()
+    {
+        const double step = 3.14/22050;
+        double value = 0;
+        for (int j = 0; j < 3; j++)
+        {
+            value = 0;
+            for (int i = 0; i < 44100; i++)
+            {
+                validAudioData.push_back(sin(value));
+                value += step;
+            }
+        }
+    }
+    void getValidVideoData()
+    {
+        Mat img(height, width, videoType);
+        for (int i = 0; i < numberOfFrames; ++i)
+        {
+            generateFrame(i, numberOfFrames, img);
+            validVideoData.push_back(img);
+        }
+    }
+    void getFileData()
+    {
+        ASSERT_TRUE(cap.open(findDataFile(root + fileName + "." + format), backend.second, params));
+        const int audioBaseIndex = static_cast<int>(cap.get(cv::CAP_PROP_AUDIO_BASE_INDEX));
+        double f = 0;
+        int samplePerSecond = (int)cap.get(CAP_PROP_AUDIO_SAMPLES_PER_SECOND);
+        ASSERT_EQ(samplePerSecond, 44100);
```

Comment:
> ASSERT_EQ

The first argument is "expected" value. See Google Test documentation for _EQ checks.
Arguments should be swapped.

(there are several similar problems)